### PR TITLE
Feature/f2p 47

### PR DESCRIPTION
**Pull request by n3op2**
### What
For story - [F2P-47](https://digicatapult.atlassian.net/browse/F2P-47)

- added cypress scripts `npm run test:dev, npm t, npm run test:integration`
- updated ignore files
- tests are included in github workflows now
- configuration to work with webpack

#### Validation
- [ ] - Confirm in github actions

#### Output 

Component tests
```
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        9.5.3                                                                          │
  │ Browser:        Electron 94 (headless)                                                         │
  │ Node Version:   v16.13.2 (/Users/PMichelevicius/.nvm/versions/node/v16.13.2/bin/node)          │
  │                 m                                                                              │
  │ Specs:          1 found (__tests__/App.test.js)                                                │
  │ Searched:       src/**/*                                                                       │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  __tests__/App.test.js                                                           (1 of 1)
assets by path static/ 3.04 MiB
  assets by path static/media/ 929 KiB
    asset static/media/fontawesome-webfont.c1e38fd9e0e74ba58f7a.svg 434 KiB [emitted] [immutable] [from: src/fonts/fontawesome-webfont.svg?v=4.7.0] (auxiliary name: spec-0)
    asset static/media/fontawesome-webfont.8b43027f47b20503057d.eot 162 KiB [emitted] [immutable] [from: src/fonts/fontawesome-webfont.eot?v=4.7.0] (auxiliary name: spec-0)
    asset static/media/fontawesome-webfont.1e59d2330b4c6deb84b3.ttf 162 KiB [emitted] [immutable] [from: src/fonts/fontawesome-webfont.ttf?v=4.7.0] (auxiliary name: spec-0)
    asset static/media/fontawesome-webfont.f691f37e57f04c152e23.woff 95.7 KiB [emitted] [immutable] [from: src/fonts/fontawesome-webfont.woff?v=4.7.0] (auxiliary name: spec-0)
    asset static/media/fontawesome-webfont.20fd1704ea223900efa9.woff2 75.4 KiB [emitted] [immutable] [from: src/fonts/fontawesome-webfont.woff2?v=4.7.0] (auxiliary name: spec-0)
  assets by path static/js/*.js 2.13 MiB
    asset static/js/vendors-node_modules_cypress_react_dist_cypress-react_esm-bundler_js-node_modules_buffer_inde-0977c6.chunk.js 1.35 MiB [emitted] (id hint: vendors) 1 related asset
    asset static/js/spec-0.chunk.js 797 KiB [emitted] (name: spec-0) 1 related asset
    asset static/js/cypress_support_index_js.chunk.js 2.42 KiB [emitted] 1 related asset
asset main.js 227 KiB [emitted] (name: main) (id hint: vendors) 1 related asset
asset asset-manifest.json 1.5 KiB [emitted]
asset index.html 345 bytes [emitted]
cached modules 2.2 MiB (javascript) 1.06 MiB (asset) [cached] 140 modules
runtime modules 7.2 KiB 11 modules
javascript modules 2.05 KiB
  ./node_modules/@cypress/webpack-dev-server/dist/loader.js!./node_modules/@cypress/webpack-dev-server/dist/browser.js 886 bytes [not cacheable] [built] [code generated]
  ./src/__tests__/App.test.js 577 bytes [built] [code generated]
  ./src/App.js 632 bytes [built]
webpack 5.70.0 compiled successfully in 1887 ms


  App.js
    ✓ renders App component (118ms)


  1 passing (139ms)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     0 seconds                                                                        │
  │ Spec Ran:     __tests__/App.test.js                                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  __tests__/App.test.js                    135ms        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        135ms        1        1        -        -        -  

~/workspace/veritable-verifier (feature/f2p-47) $ git add -p
```

Integration
```
~/workspace/veritable-verifier (feature/f2p-47) $ npm run test:integration

> veritable-verifier@0.1.2 test:integration
> npx cypress run

^[[A
====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        9.5.3                                                                          │
  │ Browser:        Electron 94 (headless)                                                         │
  │ Node Version:   v16.13.2 (/Users/PMichelevicius/.nvm/versions/node/v16.13.2/bin/node)          │
  │                 m                                                                              │
  │ Specs:          1 found (connects-to-agent.spec.js)                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  connects-to-agent.spec.js                                                       (1 of 1)
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating


  Integration tests for Holder persona
    happy path
      ✓ renders DOM (929ms)
      if agent does not respond
        ✓ renders a modal along with the error message (252ms)
      connection to agent
        ✓ retrieves agent 's status from `/status?` endpoint (307ms)
        ✓ gets list of connections from '/connections?' endpoint (1921ms)


  4 passing (4s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        4                                                                                │
  │ Passing:      4                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     4 seconds                                                                        │
  │ Spec Ran:     connects-to-agent.spec.js                                                        │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  connects-to-agent.spec.js                00:04        4        4        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:04        4        4        -        -        -  

~/workspace/veritable-verifier (feature/f2p-47) $ npm run test:integration
~/workspace/veritable-verifier (feature/f2p-47) $ 
```

### CODE REVIEW

File: .github/workflows/react-lint-testing.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: TestLinting
+name: Branch
```
Comment (jonmattgray) [inline .github/workflows/react-lint-testing.yml:1]: rename the file to `branch` as well like in `holder`?

Comment (n3op2) [reply .github/workflows/react-lint-testing.yml:1]: updated in holder to be lint and testing:) -> https://github.com/digicatapult/veritable-holder/pull/7/files

File: .github/workflows/react-lint-testing.yml [hunk 2]
```diff
@@ -10,28 +10,23 @@ on:
...
-        run: |
-          npm ci
-
+        run: npm ci
       - name: Run lint
-        run: |
-          npm run lint
-
+        run: npm run lint
+      - name: Run component tests
```
Comment (jonmattgray) [inline .github/workflows/react-lint-testing.yml:26]: could add integration tests to `release.yml` as well (I'm just copying what you did on `holder`). Also README how to run tests

Comment (n3op2) [reply .github/workflows/react-lint-testing.yml:26]: oh... pretty sure I did copy, might have forgotten to push, uno momento, thanks for flagging.

**Review by dc-andysign — APPROVED** [review]
Looks good to me

**Review by dc-andysign — DISMISSED** [review]
Looks good to me

**Review by jonmattgray — DISMISSED** [review]

**Review by jonmattgray — APPROVED** [review]

**Pull request merged**